PowerShell/ScubaGear/Testing/Unit/PowerShell/RunRego/Run-Rego.Tests.ps1

```diff
@@ -0,0 +1,75 @@
+                sharepoint    = "SharePoint";
+                onedrive      = "OneDrive";
+            }
+            [Diagnostics.CodeAnalysis.SuppressMessageAttribute('PSUseDeclaredVarsMoreThanAssignments', 'RegoParams')]
+            $RegoParams = @{
+                'InputFile' = Join-Path -Path $PSScriptRoot -ChildPath "./RunRegoStubs/ProviderSettingsExport.json";
+                'OPAPath'   = Join-Path -Path $env:USERPROFILE -ChildPath ".scubagear/Tools";
+            }
+        }
+        It 'Runs the AAD Rego on a Provider JSON and returns a TestResults object' {
+            $Product = 'aad'
+            $RegoParams += @{
+                'RegoFile'    = Join-Path -Path $PSScriptRoot -ChildPath "../../../../Rego/$($ArgToProd[$Product])Config.rego";
+                'PackageName' = $Product;
+            }
+            Invoke-Rego @RegoParams | Should -Not -Be $null
+        }
+        It 'Runs the Defender Rego on a Provider JSON and returns a TestResults object' {
+            $Product = 'defender'
+            $RegoParams += @{
+                'RegoFile'    = Join-Path -Path $PSScriptRoot -ChildPath "../../../../Rego/$($ArgToProd[$Product])Config.rego";
+                'PackageName' = $Product;
+            }
+            Invoke-Rego @RegoParams | Should -Not -Be $null
+        }
+        It 'Runs the EXO Rego on a Provider JSON and returns a TestResults object' {
+            $Product = 'exo'
+            $RegoParams += @{
+                'RegoFile'    = Join-Path -Path $PSScriptRoot -ChildPath "../../../../Rego/$($ArgToProd[$Product])Config.rego";
+                'PackageName' = $Product;
+            }
+            Invoke-Rego @RegoParams | Should -Not -Be $null
+        }
+        It 'Runs the PowerPlatform Rego on a Provider JSON and returns a TestResults object' {
+            $Product = 'powerplatform'
+            $RegoParams += @{
+                'RegoFile'    = Join-Path -Path $PSScriptRoot -ChildPath "../../../../Rego/$($ArgToProd[$Product])Config.rego";
+                'PackageName' = $Product;
+            }
+            Invoke-Rego @RegoParams | Should -Not -Be $null
+        }
+        It 'Runs the SharePoint Rego on a Provider JSON and returns a TestResults object' {
+            $Product = 'sharepoint'
+            $RegoParams += @{
+                'RegoFile'    = Join-Path -Path $PSScriptRoot -ChildPath "../../../../Rego/$($ArgToProd[$Product])Config.rego";
+                'PackageName' = $Product;
+            }
+            Invoke-Rego @RegoParams | Should -Not -Be $null
+        }
+        It 'Runs the Teams Rego on a Provider JSON and returns a TestResults object' {
+            $Product = 'teams'
+            $RegoParams += @{
+                'RegoFile'    = Join-Path -Path $PSScriptRoot -ChildPath "../../../../Rego/$($ArgToProd[$Product])Config.rego";
+                'PackageName' = $Product;
+            }
+            Invoke-Rego @RegoParams | Should -Not -Be $null
```

Comment:
This may be me being nit picky, but can this be refactored to a function, very repetitive code.

Reply:
I took a different approach to the refactoring.  Look at see if it answers the comment.  

Results look good. 
![image](https://github.com/cisagov/ScubaGear/assets/7025068/e4eff22f-774c-4e0a-8169-214813f7fd29)